eth_utils/abi.py

```diff
@@ -1,68 +1,784 @@
+            ...         "type": "function",
+            ...     },
+            ...     {
+            ...         "constant": False,
+            ...         "inputs": [
+            ...             {"name": "a", "type": "uint256"},
+            ...         ],
+            ...         "name": "func_2",
+            ...         "outputs": [],
+            ...         "type": "function",
+            ...     },
+            ...     {
+            ...         "constant": False,
+            ...         "inputs": [
+            ...             {"name": "a", "type": "uint256"},
+            ...             {"name": "b", "type": "uint256"},
+            ...         ],
+            ...         "name": "func_3",
+            ...         "outputs": [],
+            ...         "type": "function",
+            ...     },
+            ...     {
+            ...         "constant": False,
+            ...         "inputs": [
+            ...             {"name": "a", "type": "uint256"},
+            ...             {"name": "b", "type": "uint256"},
+            ...             {"name": "c", "type": "uint256"},
+            ...         ],
+            ...         "name": "func_4",
+            ...         "outputs": [],
+            ...         "type": "function",
+            ...     },
+            ... ]
+            >>> filter_abi_by_name("func_1", abi)
+            [{'constant': False, 'inputs': [], 'name': 'func_1', 'outputs': [], \
+'type': 'function'}]
+    """
+    return [
+        abi
+        for abi in contract_abi
+        if (
+            (
+                abi["type"] == "function"
+                or abi["type"] == "event"
+                or abi["type"] == "error"
+            )
+            and abi["name"] == name
+        )
+    ]
+
+
+def filter_abi_by_type(type: str, contract_abi: ABI) -> Sequence[ABIElement]:
+    """
+    Return a list of each ``ABIElement`` that is of type ``type``.
+
+    :param type: Type of ABI element to filter by.
+    :type type: `str`
+    :param contract_abi: Contract ABI.
+    :type contract_abi: `ABI`
+    :return: List of ABI elements of the specified type.
+    :rtype: `Sequence[ABIElement]`
+
+    .. doctest::
+
+        >>> from eth_utils import filter_abi_by_type
+        >>> abi = [
+        ...   {"type": "function", "name": "myFunction", "inputs": [], "outputs": []},
+        ...   {"type": "function", "name": "myFunction2", "inputs": [], "outputs": []},
+        ...   {"type": "event", "name": "MyEvent", "inputs": []}
+        ... ]
+        >>> filter_abi_by_type("function", abi)
+        [{'type': 'function', 'name': 'myFunction', 'inputs': [], 'outputs': []}, \
+{'type': 'function', 'name': 'myFunction2', 'inputs': [], 'outputs': []}]
+    """
+    return [abi for abi in contract_abi if abi["type"] == type]
+
+
+def get_all_function_abis(contract_abi: ABI) -> Sequence[ABIFunction]:
+    """
+    Return interfaces for each function in the contract ABI.
+
+    :param abi: Contract ABI.
```

Comment:
This should match the arg name in the function signature.

```suggestion
    :param contract_abi: Contract ABI.
```